cli/kata-check_amd64.go

```diff
@@ -7,52 +7,120 @@ package main
 
 import (
 	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"log"
+	"strings"
 )
 
 const (
 	cpuFlagsTag        = genericCPUFlagsTag
 	archCPUVendorField = genericCPUVendorField
 	archCPUModelField  = genericCPUModelField
+	archGenuineIntel   = "GenuineIntel"
+	archAuthenticAMD   = "AuthenticAMD"
 )
 
+// CPU types
+const (
+	CPUtypeIntel   = 0
+	CPUtypeAMD     = 1
+	CPUtypeUnknown = -1
+)
+
+// CPUtype save the CPU type
+var CPUtype int
+
 // archRequiredCPUFlags maps a CPU flag value to search for and a
 // human-readable description of that value.
-var archRequiredCPUFlags = map[string]string{
-	"vmx":    "Virtualization support",
-	"lm":     "64Bit CPU",
-	"sse4_1": "SSE4.1",
-}
+var archRequiredCPUFlags map[string]string
 
 // archRequiredCPUAttribs maps a CPU (non-CPU flag) attribute value to search for
 // and a human-readable description of that value.
-var archRequiredCPUAttribs = map[string]string{
-	"GenuineIntel": "Intel Architecture CPU",
-}
+var archRequiredCPUAttribs map[string]string
 
 // archRequiredKernelModules maps a required module name to a human-readable
 // description of the modules functionality and an optional list of
 // required module parameters.
-var archRequiredKernelModules = map[string]kernelModule{
-	"kvm": {
-		desc: "Kernel-based Virtual Machine",
-	},
-	"kvm_intel": {
-		desc: "Intel KVM",
-		parameters: map[string]string{
-			"nested": "Y",
-			// "VMX Unrestricted mode support". This is used
-			// as a heuristic to determine if the system is
-			// "new enough" to run a Kata Container
-			// (atleast a Westmere).
-			"unrestricted_guest": "Y",
-		},
-	},
-	"vhost": {
-		desc: "Host kernel accelerator for virtio",
-	},
-	"vhost_net": {
-		desc: "Host kernel accelerator for virtio network",
-	},
+var archRequiredKernelModules map[string]kernelModule
+
+func init() {
+	CPUtype = getCPUtype()
+
+	if CPUtype == CPUtypeIntel {
+		archRequiredCPUFlags = map[string]string{
+			"vmx":    "Virtualization support",
+			"lm":     "64Bit CPU",
+			"sse4_1": "SSE4.1",
+		}
+		archRequiredCPUAttribs = map[string]string{
+			archGenuineIntel: "Intel Architecture CPU",
+		}
+		archRequiredKernelModules = map[string]kernelModule{
+			"kvm": {
+				desc: "Kernel-based Virtual Machine",
```

Comment:
There are now quite a few duplicated strings so might be good to define them as private const variables to avoid the duplication. That would also make maintenance easier.